### PR TITLE
#10671: API to get chip location

### PR DESCRIPTION
**Pull request by jnie-TT**
### Ticket
[Link to Github Issue](https://github.com/tenstorrent/tt-metal/issues/10671)

### What's changed
API added to Device and Cluster to get chip location

### CODE REVIEW

File: tt_metal/llrt/tt_cluster.cpp
Comment (nsmithtt) [reply tt_metal/llrt/tt_cluster.cpp:345]: I'm also thinking that typically the call to `get_chip_location` is only something an app (like ttrt) will do offline and infrequently so not too worried about overhead.

Comment (jnie-TT) [reply tt_metal/llrt/tt_cluster.cpp]: Hey @nsmithtt great point, @aliuTT brought this up too - it's not expensive so I'll just initialize it as a local variable. @aliuTT also suggested we should directly query this cluster API instead of adding it to device. Pushed an update that does that as well.

File: tt_metal/llrt/tt_cluster.cpp
Comment (jnie-TT) [reply tt_metal/llrt/tt_cluster.cpp]: Assigned it as reference. Still want to catch the case when the chip isn't in the map instead of debugging map_base::at errors.

**Review by nsmithtt — APPROVED** [review]

**Review by aliuTT — APPROVED** [review]

**Comment by davorchap** [comment]
@jnie-TT  and @jnie-TT: this API is a layer violation, it's a backdoor, and bypasses ttnn and Metal-Runtime APIs.

if you require a new API , please file an API request:
*  use case
* why a new a new API is needed , ie why current API isn’t sufficient

Metalium stack has these layers:

**Comment by davorchap** [comment]
In addition to filing a request, the process for fulling a request is:
* it would be triaged to the proper Board (ttnn, Metal, UMD, etc)
* proper CODEOWNERS (@abhullar-tt , @pgkeller , @aliuTT  @ubcheema , @davorchap )
* assigned as P0 / P1 / P2 
* and then scheduled for work

**Comment by jnie-TT** [comment]
@davorchap I talked to @aliuTT and we agreed to add it as a cluster API and not directly expose it in upper layers. Apologies if this caused any inconvenience, would it be desirable for me to revert this and file an API request? @nsmithtt FYI

**Comment by aliuTT** [comment]
Please revert for now. This commit shouldn't be addressing any needs from MLIR, since we should never be bypassing Device apis anyway. Apologies Jackson, I jumped the gun on approving this PR. We should sync first on what APIs to expose even at the ttCluster level that isn't user facing.

**Comment by jnie-TT** [comment]
@aliuTT no worries, thanks. Created a PR to revert this [here](https://github.com/tenstorrent/tt-metal/pull/10744)

**Comment by tt-rkim** [comment]
Note that there are no CODEOWNERS for this section of the runtime because runtime team previously complained that PRs took too long.

I'm going to talk to the runtime team about putting more files under purview, but this also means more PRs will need approval.

**Pull request merged**
